### PR TITLE
[ch9234] t2f simplify workflow

### PR DESCRIPTION
**Pull request by domdinicola**
[ch9234]

### CODE REVIEW

File: src/etools/applications/t2f/helpers/misc.py [hunk 1]
```diff
@@ -1,10 +0,0 @@
-
-from etools.applications.t2f.models import Travel
-
-
-def get_open_travels_for_check(obj):
```
Comment (domdinicola) [inline src/etools/applications/t2f/helpers/misc.py]: these status don't exist anymore

File: src/etools/applications/t2f/management/commands/_private.py [hunk 2]
```diff
@@ -173,26 +173,15 @@ def set_up_traveler(self):
         qs = TravelPermission.objects.filter(user_type=UserTypes.TRAVELER)
 
         self.revoke_edit(qs.filter(model='travel', field='traveler'))
-        status_where_hide = [Travel.PLANNED,
```
Comment (domdinicola) [inline src/etools/applications/t2f/management/commands/_private.py]: this command has broken import based on non-existing models

File: src/etools/applications/t2f/models.py [hunk 3]
```diff
@@ -131,13 +125,14 @@ class Travel(models.Model):
...
         (COMPLETED, _('Completed')),
         (CANCELLED, _('Cancelled')),
-        (SENT_FOR_PAYMENT, _('Sent for payment')),
-        (CERTIFICATION_SUBMITTED, _('Certification submitted')),
-        (CERTIFICATION_APPROVED, _('Certification approved')),
-        (CERTIFICATION_REJECTED, _('Certification rejected')),
-        (CERTIFIED, _('Certified')),
         (COMPLETED, _('Completed')),
     )
+    SUBMIT_FOR_APPROVAL = 'submit_for_approval'
```
Comment (domdinicola) [inline src/etools/applications/t2f/models.py]: i like better to use constants for transaction name

File: src/etools/applications/t2f/models.py [hunk 4]
```diff
@@ -280,20 +275,6 @@ def has_supervisor(self):
             raise TransitionError('Travel has no supervisor defined. Please select one.')
         return True
 
-    def check_travel_count(self):
-        from etools.applications.t2f.helpers.misc import get_open_travels_for_check
```
Comment (domdinicola) [inline src/etools/applications/t2f/models.py]: get_open_travels_for_check has been removed

File: src/etools/applications/t2f/models.py [hunk 5]
```diff
@@ -357,56 +333,8 @@ def cancel(self):
     def plan(self):
         pass
 
-    @send_for_payment_threshold_decorator
-    @transition(status, source=[APPROVED, SENT_FOR_PAYMENT, CERTIFIED], target=SENT_FOR_PAYMENT)
```
Comment (domdinicola) [inline src/etools/applications/t2f/models.py]: target status has been deleted

File: src/etools/applications/t2f/models.py [hunk 6]
```diff
@@ -357,56 +333,8 @@ def cancel(self):
...
-
-        # If invoicing is turned off, don't send a mail
-        if settings.DISABLE_INVOICING and self.ta_required:
-            return
-
-        self.send_notification_email('Travel #{} sent for payment.'.format(self.reference_number),
-                                     self.traveler.email,
-                                     'emails/sent_for_payment.html')
-
-    @transition(status, source=[SENT_FOR_PAYMENT, CERTIFICATION_REJECTED, CERTIFIED],
```
Comment (domdinicola) [inline src/etools/applications/t2f/models.py]: target status has been deleted

File: src/etools/applications/t2f/models.py [hunk 7]
```diff
@@ -357,56 +333,8 @@ def cancel(self):
...
-                target=CERTIFICATION_SUBMITTED,
-                conditions=[check_pending_invoices])
-    def submit_certificate(self):
-        self.send_notification_email('Travel #{} certification was submitted.'.format(self.reference_number),
-                                     self.supervisor.email,
-                                     'emails/certificate_submitted.html')
-
-    @transition(status, source=[CERTIFICATION_SUBMITTED], target=CERTIFICATION_APPROVED)
```
Comment (domdinicola) [inline src/etools/applications/t2f/models.py]: target status has been deleted

File: src/etools/applications/t2f/models.py [hunk 8]
```diff
@@ -357,56 +333,8 @@ def cancel(self):
...
-    def approve_certificate(self):
-        self.send_notification_email('Travel #{} certification was approved.'.format(self.reference_number),
-                                     self.traveler.email,
-                                     'emails/certificate_approved.html')
-
-    @transition(status, source=[CERTIFICATION_APPROVED, CERTIFICATION_SUBMITTED],
```
Comment (domdinicola) [inline src/etools/applications/t2f/models.py]: target status has been deleted

File: src/etools/applications/t2f/models.py [hunk 9]
```diff
@@ -357,56 +333,8 @@ def cancel(self):
...
-                target=CERTIFICATION_REJECTED)
-    def reject_certificate(self):
-        self.send_notification_email('Travel #{} certification was rejected.'.format(self.reference_number),
-                                     self.traveler.email,
-                                     'emails/certificate_rejected.html')
-
-    @mark_as_certified_or_completed_threshold_decorator
```
Comment (domdinicola) [inline src/etools/applications/t2f/models.py]: target status has been deleted

File: src/etools/applications/t2f/tests/test_overlapping_trips.py [hunk 10]
```diff
@@ -165,12 +164,6 @@ def test_almost_overlapping_trips(self):
         travel.approve()
```
Comment (reinbach) [inline src/etools/applications/t2f/tests/test_overlapping_trips.py]: Are these statements still needed in the test?

File: src/etools/applications/t2f/urls.py [hunk 11]
```diff
@@ -45,8 +44,7 @@
 invoices_details = InvoiceViewSet.as_view({'get': 'retrieve'})
 
 details_state_changes_pattern = r'^(?P<transition_name>submit_for_approval|approve|reject|cancel|plan|' \
-                                r'send_for_payment|submit_certificate|approve_certificate|reject_certificate|' \
-                                r'mark_as_certified|mark_as_completed)/$'
+                                r'mark_as_completed)/$'
```
Comment (reinbach) [inline src/etools/applications/t2f/urls.py]: To keep things DRY, maybe try the following;

`details_state_changes_pattern = r"^(?P<transition_name>{})/$".format("|".join([s[0] for s in Travel.CHOICES]))`

File: src/etools/applications/t2f/tests/test_overlapping_trips.py (hunk 10 above)
Comment (domdinicola) [reply src/etools/applications/t2f/tests/test_overlapping_trips.py]: we can get rid of them

File: src/etools/applications/t2f/models.py [hunk 12]
```diff
@@ -265,10 +257,6 @@ def check_completed_from_planned(self):
         return True
 
     def check_pending_invoices(self):
```
Comment (ntrncic) [inline src/etools/applications/t2f/models.py]: we can remove this whole method.

File: src/etools/applications/t2f/models.py (hunk 3 above)
Comment (ntrncic) [reply src/etools/applications/t2f/models.py]: whatever floats your boat

File: src/etools/applications/t2f/tests/test_state_machine.py [hunk 13]
```diff
@@ -138,161 +120,25 @@ def test_state_machine_flow(self):
...
-        data['report'] = 'Something'
-        response = self.forced_auth_req('post', reverse('t2f:travels:details:state_change',
-                                                        kwargs={'travel_pk': travel_id,
-                                                                'transition_name': 'mark_as_completed'}),
-                                        data=data, user=self.traveler)
-        response_json = json.loads(response.rendered_content)
-        self.assertEqual(response_json['status'], Travel.COMPLETED)
-
-    @override_settings(DISABLE_INVOICING=True)
-    def test_state_machine_flow_invoice_disabled(self):
```
Comment (ntrncic) [inline src/etools/applications/t2f/tests/test_state_machine.py]: since invoice is disabled can't we run these tests. opr modify them to run normally.

**Pull request merged**